### PR TITLE
Apply media padding to Gallery Material Design demos

### LINKED ISSUE #13594: iPhone X: update Flutter Gallery demo to use safe area insets
The Gallery demo requires a set of tweaks to apply media padding (via SafeArea or otherwise) to adapt well to the iPhone X and avoid placing content under the sensor housing notch.

### PR DESCRIPTION
**Pull request by cbracken**
Applies media padding (e.g. iPhone X safe area insets) to the Material Design demos in the Gallery.

Covers the following demos:
* Buttons (via the TabbedComponentDemoScaffold change)
* Cards
* Expansion panels
* Grid list
* Icons
* Images (via the TabbedComponentDemoScaffold change)
* Page Selector
* Progress Indicator
* Scrollable tabs
* Selection controls (via the TabbedComponentDemoScaffold change)
* Snack bar
* Tabs
* Text fields
* Tooltips

Fixes #13594

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/material/cards_demo.dart [hunk 1]
```diff
@@ -60,85 +60,89 @@ class TravelDestinationItem extends StatelessWidget {
...
-                children: <Widget>[
-                  new Positioned.fill(
-                    child: new Image.asset(
-                      destination.assetName,
-                      package: destination.assetPackage,
-                      fit: BoxFit.cover,
+    return new SafeArea(
+      top: false,
+      bottom: false,
+      child: new Container(
```
Comment (cbracken) [inline examples/flutter_gallery/lib/demo/material/cards_demo.dart]: Lines 63-66 are the only non-indent change in this file.

File: examples/flutter_gallery/lib/demo/material/date_and_time_picker_demo.dart [hunk 2]
```diff
@@ -131,74 +131,78 @@ class _DateAndTimePickerDemoState extends State<DateAndTimePickerDemo> {
...
-        child: new ListView(
-          padding: const EdgeInsets.all(16.0),
-          children: <Widget>[
-            new TextField(
-              decoration: const InputDecoration(
-                labelText: 'Event name',
+        child: new SafeArea(
+          top: false,
+          bottom: false,
+          child: new ListView(
```
Comment (cbracken) [inline examples/flutter_gallery/lib/demo/material/date_and_time_picker_demo.dart]: Note: lines 134-137 are the only non-indent change in this file.

File: examples/flutter_gallery/lib/demo/material/expansion_panels_demo.dart [hunk 3]
```diff
@@ -336,24 +336,28 @@ class _ExpansionPanelsDemoState extends State<ExpasionPanelsDemo> {
...
-                body: item.builder(item)
-              );
-            }).toList()
-          )
-        )
-      )
+        child: new SafeArea(
+          top: false,
+          bottom: false,
+          child: new Container(
```
Comment (cbracken) [inline examples/flutter_gallery/lib/demo/material/expansion_panels_demo.dart]: Note: lines 339-342 are the only non-indent change in this file.

File: examples/flutter_gallery/lib/demo/material/grid_list_demo.dart [hunk 4]
```diff
@@ -363,27 +363,31 @@ class GridListDemoState extends State<GridListDemo> {
...
-                );
-              }).toList(),
-            )
-          )
-        ]
-      )
+            child: new SafeArea(
+              top: false,
+              bottom: false,
+              child: new GridView.count(
```
Comment (cbracken) [inline examples/flutter_gallery/lib/demo/material/grid_list_demo.dart]: Note: lines 366-369 are the only non-indent change in this file.

File: examples/flutter_gallery/lib/demo/material/page_selector_demo.dart [hunk 5]
```diff
@@ -19,45 +19,49 @@ class _PageSelector extends StatelessWidget {
...
-                key: new ObjectKey(icon),
-                padding: const EdgeInsets.all(12.0),
-                child: new Card(
-                  child: new Center(
-                    child: new Icon(icon, size: 128.0, color: color)
-                  ),
+    return new SafeArea(
+      top: false,
+      bottom: false,
+      child: new Column(
```
Comment (cbracken) [inline examples/flutter_gallery/lib/demo/material/page_selector_demo.dart]: Note: lines 22-25 are the only non-indent change in this file.

File: examples/flutter_gallery/lib/demo/material/progress_indicator_demo.dart [hunk 6]
```diff
@@ -107,17 +107,21 @@ class _ProgressIndicatorDemoState extends State<ProgressIndicatorDemo> with Sing
...
-                )
-              )
-            )
-          )
-        )
-      )
+              child: new SafeArea(
+                top: false,
+                bottom: false,
+                child: new Container(
```
Comment (cbracken) [inline examples/flutter_gallery/lib/demo/material/progress_indicator_demo.dart]: Note: lines 110-113 are the only non-indent change in this file.

File: examples/flutter_gallery/lib/demo/material/snack_bar_demo.dart [hunk 7]
```diff
@@ -29,39 +29,43 @@ class _SnackBarDemoState extends State<SnackBarDemo> {
...
-        return new Container(
-          margin: const EdgeInsets.symmetric(vertical: 12.0),
-          child: child
-        );
-      })
-      .toList()
+    return new SafeArea(
+      top: false,
+      bottom: false,
+      child: new ListView(
```
Comment (cbracken) [inline examples/flutter_gallery/lib/demo/material/snack_bar_demo.dart]: Note: lines 33-35 are the only non-indent change in this file.

File: examples/flutter_gallery/lib/demo/material/text_form_field_demo.dart [hunk 8]
```diff
@@ -108,98 +108,102 @@ class TextFormFieldDemoState extends State<TextFormFieldDemo> {
...
-          children: <Widget>[
-            new TextFormField(
-              decoration: const InputDecoration(
-                icon: const Icon(Icons.person),
-                hintText: 'What do people call you?',
-                labelText: 'Name *',
+      body: new SafeArea(
+        top: false,
+        bottom: false,
+        child: new Form(
```
Comment (cbracken) [inline examples/flutter_gallery/lib/demo/material/text_form_field_demo.dart]: Note: lines 111-114 are the only non-indent change in this file.

File: examples/flutter_gallery/lib/demo/material/tooltip_demo.dart [hunk 9]
```diff
@@ -22,44 +22,48 @@ class TooltipDemo extends StatelessWidget {
...
-                  tooltip: 'Place a phone call',
-                  onPressed: () {
-                    Scaffold.of(context).showSnackBar(const SnackBar(
-                       content: const Text('That was an ordinary tap.')
-                    ));
-                  }
+          return new SafeArea(
+            top: false,
+            bottom: false,
+            child: new ListView(
```
Comment (cbracken) [inline examples/flutter_gallery/lib/demo/material/tooltip_demo.dart]: Note: lines 25-28 are the only non-indent change in this file.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**